abn/watcher/informer.go

```diff
@@ -0,0 +1,82 @@
+package watcher
+
+// informer.go - informer(s) to watch desired resources/namespaces
+
+import (
+	"github.com/iter8-tools/iter8/abn/application"
+	"github.com/iter8-tools/iter8/base/log"
+	"github.com/iter8-tools/iter8/driver"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic/dynamicinformer"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+)
+
+// MultiInformer holds all of the informers for all namespaces
+type MultiInformer struct {
+	informersByKey             map[string]informers.GenericInformer
+	informerFactroyByNamespace map[string]dynamicinformer.DynamicSharedInformerFactory
+}
+
+// addInformer adds an informer
+func (informer *MultiInformer) addInformer(ns string, gvr schema.GroupVersionResource) {
+	informer.informersByKey[key(ns, gvr)] = informer.informerFactroyByNamespace[ns].ForResource(gvr)
+}
```

Comment:
A result of team review was a recommendation to explore the watch api instead of the informer api.

Reply:
propose we continue to use informers; see comparison https://iter8-tools.slack.com/archives/C01NV4T21HU/p1660598419750459